internal/xds/rbac/rbac_engine_test.go

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rbac
+
+import (
+	"testing"
+
+	v3rbacpb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v3"
+	v3routepb "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	v3matcherpb "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
+	"google.golang.org/grpc/internal/grpctest"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+// TestRBACEngine tests the RBAC Engine by configuring the engine in different
+// scenarios. After configuring the engine in a certain way, this test pings the
+// engine with different kinds of data representing incoming RPC's, and verifies
+// that it works as expected.
+func (s) TestRBACEngine(t *testing.T) {
+	tests := []struct {
+		name        string
+		rbacConfig  *v3rbacpb.RBAC
+		rbacQueries []struct {
+			evaluateArgs           *EvaluateArgs
+			wantMatchingPolicyName string
+		}
+	}{
+		// TestSuccessCaseAnyMatch tests an RBAC Engine instantiated with a
+		// config with a policy with any rules for both permissions and
+		// principals, meaning that any data about incoming RPC's that the RBAC
+		// Engine is queried with should match that policy.
+		{name: "TestSuccessCaseAnyMatch",
```

Comment:
Please start the struct literal initializations with a line which has just the open parentheses. That will ensure that all fields are indented at the same level.
Here and everywhere else in this test.

Reply:
Done.